modules/ld/shacl-validator/src/main/scala/ch/epfl/bluebrain/nexus/commons/shacl/validator/ShaclValidator.scala

```diff
@@ -35,6 +35,27 @@ final class ShaclValidator[F[_]](importResolver: ImportResolver[F])(implicit F:
   private val schemaEngine     = ShaclexSchema.empty.name
   private val triggerMode      = TargetDeclarations.name
 
+  /**
+    * Validates ''data'' in its Jena model representation against the specified ''schema''.  It produces a
+    * ''ValidationReport'' in the ''F[_]'' context.
+    *
+    * @param schema the shacl schema instance against which data is validated
+    * @param data   the data to be validated
+    * @return a ''ValidationReport'' in the ''F[_]'' context
+    */
+  def apply(schema: Schema, data: RDFAsJenaModel): F[ValidationReport] =
```

Comment:
Don't we need data to be `rdf.Graph` (our graph representation) instead of this `RDFAsJenaModel ` ?